src/DataType/UserRequests.ts

```diff
@@ -0,0 +1,66 @@
+import localforage from 'localforage';
+import { rethrow } from '../Utility/errors';
+import Privacy, { PRIVACY_ACTIONS } from '../Utility/Privacy';
+
+export type UserRequest = {
+    reference: string;
+    date: string;
+    type: 'access' | 'erasure' | 'rectification' | 'objection' | 'custom';
+    response_type?: 'admonition' | 'complaint';
+    slug: string;
+    recipient: string;
+    email: string;
+    via: 'fax' | 'letter' | 'email';
+};
+
+export class UserRequests {
+    localforage_instance?: typeof localforage;
+
+    constructor() {
+        if (Privacy.isAllowed(PRIVACY_ACTIONS.SAVE_MY_REQUESTS)) {
+            this.localforage_instance = localforage.createInstance({
```

Comment:
What if the constructor is called, if `Privacy.isAllowed(PRIVACY_ACTIONS.SAVE_MY_REQUESTS)` is false, but the state later changes and there is no `localforage_instance` available?